src/org/mozilla/javascript/resources/Messages.properties

```diff
@@ -519,6 +519,9 @@ msg.destruct.default.vals =\
 msg.no.old.octal.strict =\
     Old octal numbers prohibited in strict mode.
 
+msg.no.old.octal.bigint =\
+    Old octal numbers prohibited in BigInt.
+
```

Comment:
I don't think this message is necessary. Below is the current behavior, and it is only triggered in strict mode, so I think the existing message is good enough, and you don't need a special case for bigint in the parser.

```
js> (function() {return 013})()
11
js> (function() {"use strict"; return 013})()
js: "<stdin>", line 24: Old octal numbers prohibited in strict mode.
js: (function() {"use strict"; return 013})()
js: .....................................^
js: "<stdin>", line 24: missing } after function body
js: (function() {"use strict"; return 013})()
js: ........................................^
js: "<stdin>", line 24: Compilation produced 2 syntax errors.

js> (function() {return 013n})() // bigint conversion to octal when not in strict mode
11
js> (function() {"use strict"; return 013n})()
js: "<stdin>", line 26: Old octal numbers prohibited in BigInt.
js: (function() {"use strict"; return 013n})()
js: ......................................^
js: "<stdin>", line 26: missing } after function body
js: (function() {"use strict"; return 013n})()
js: .........................................^
js: "<stdin>", line 26: Compilation produced 2 syntax errors.
```

Technically, the spec doesn't allow legacy octal notation at all, but lists is as a browser extension when not in strict mode. Chrome supports it in the same way that we do for numbers, but reports a generic Syntax Error for bigints.

Node.js reports a generic Syntax Error for both numbers and bigints, even when not in strict mode, per the spec. I think we should match this behavior for language version 200+. I would open an issue for this after merging, though. I don't think it needs to stop this PR from going in.

Reply:
The rest of the messages look fine to me.

Reply:
The latest test262 had this test case.
https://github.com/tc39/test262/blob/2ee3864136747ee69401b2d266e234cdd0a95965/test/language/literals/bigint/legacy-octal-like-invalid-00n.js

Reply:
There seems to have been a mistake when I fixed the conflicts with #814.

Reply:
I think it's working well enough to merge now, and then fix after. Existing code won't break, because it previously would not parse at all.

Reply:
That's fine with me -- like we said the "123n" syntax didn't work at all before.